enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,404 @@
+  class cluster2 cluster1;
+```
+
+### API Extensions
+
+Added to the CredentialsRequest API 
+
+We outline this change using the AWS specific change, but other cloud provider specs will follow a similar pattern:
+Adding to the `spec.ProviderSpec` field ([ref](https://github.com/openshift/cloud-credential-operator/blob/38e7f96bf73036924abcb1352222c17704e4b2e0/pkg/apis/cloudcredential/v1/types_credentialsrequest.go#L53-L55)),<br>
+where `spec.ProviderSpec` would be an `AWSProviderSpec` ([ref](https://github.com/openshift/cloud-credential-operator/blob/38e7f96bf73036924abcb1352222c17704e4b2e0/pkg/apis/cloudcredential/v1/types_aws.go#L27-L31)) for AWS-directed TAT.<p>
+CredentialsRequests:
+```go
+type CredentialsRequestSpec struct {
+...
+
+// ProviderSpec contains the cloud provider specific credentials specification.
+// +kubebuilder:pruning:PreserveUnknownFields
+ProviderSpec *runtime.RawExtension `json:"providerSpec,omitempty"`
+
+...
+// CloudTokenPath (JWT token)
+// +optional
+CloudTokenPath string `json:"cloudTokenPath,omitempty"`
+}
+```
+
+and `ProviderSpec` in the AWS case would include the following:
+
+```go
+type AWSProviderSpec struct {
+...
+// STSRoleARN is the Amazon Resource Name (ARN) of an IAM Role which was created manually for the associated
+// CredentialsRequest.
+// The presence of an STSRoleARN within the AWSProviderSpec initiates creation of a secret containing IAM
+// Role details necessary for assuming the IAM Role via Amazon's Secure Token Service.
+// +optional
+STSIAMRoleARN string `json:"stsIAMRoleARN,omitempty"`
+}
+```
+
+On AWS STS, the `STSIAMRoleARN` would be something like:
+`arn:aws:iam::269733383066:role/newstscluster-openshift-logging-role-name`<br>
+This provides the role information linked by policy to access various cloud resources.
+
+`CloudTokenPath` provides the path to a JWT web token needed to access the resources.
+
+### Risks and Mitigations
+
+### Drawbacks
+
+## Design Details
+
+We have two PoC items:<br>
+How CCO changes might be implemented: https://github.com/openshift/cloud-credential-operator/pull/525
+
+How Cluster Logging Operator can be changed to work with this modified CCO: 
+https://github.com/gallettilance/cluster-logging-operator/tree/sts
+
+### Test Plan
+
+Consider the following in developing a test plan for this enhancement:
+- CCO should add unit tests for the new manner of working when in Manual mode but the 3-way test indicates running on
+  a TAT enabled cluster.
+- CCO should add e2e tests for the new manner of working when in Manual mode but the 3-way test indicates running on
+  a TAT enabled cluster.
+- CI will need to add a TAT mode for various cloud providers as a base for the e2e tests.
+- A "canary" operator should be fielded in CI with e2e tests such that we have signal on breaking changes in TAT enabled
+  clusters.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+#### Tech Preview -> GA
+#### Removing a deprecated feature
+#### Support Procedures
```

Comment:
How spread are the changes required here? How many components in the core payload does this touch? Wondering if you have considered introducing the changes behind a feature gate at first?

Reply:
They're being added to `TechPreviewNoUpgrade`. Will go make sure that the graduation criteria details that we will use the new scheme.

CCO and OLM are the only core components affected by this change beyond the TechPreviewNoUpgrade machinery.